### ISSUE TITLE
Use next-router addon instead of blatantly copy and pasting it

### ISSUE THREAD

**Issue by lifeiscontent** [RyanClementsHax/storybook-addon-next#2]
I've spent a lot of time with the next/storybook community making sure I've built something that works for them. I'd appreciate it if you could use my addon internally rather than just copy/pasting it.

It would also limit the scope of bugs for you and allow us to collaborate.

What do you think?

**Comment by RyanClementsHax** [comment]
Yeah I'd be happy to collaborate! 😃

I originally built this to repackage your addon actually, but was having problems. I can't quite remember what problems I ran into but I think it was unrelated to your addon itself. Since then I've learned a lot and am willing to revisit getting it working.

For now I just published something just to get something shipped (better done than perfect), so thanks for the feedback.

As of right now specifically, I haven't packaged your addon because I'm working on building out a feature your addon doesn't support. I'm working on building in support for older versions of nextjs using webpack magic without having the user to provide any boilerplate or interact with nextjs internals (I'm trying to go for a "works out of the box" approach with this addon). I intend to support all the versions your addon supports. I was hoping to at least get support for those versions out the door soon just to make this addon usable for most of the nextjs ecosystem (should be done soon as I have a working POC and am now just cleaning up the code). After, however, would you like to have a feature like that merged upstream to your addon?

By the way mind if I ask how you found out about this addon? I'm still amazed at how word spreads in open source especially since I only published this last week lol.

Also do u have any other feedback for this addon?

**Comment by RyanClementsHax** [comment]
Here is the pull request detailing what I had in mind:

https://github.com/RyanClementsHax/storybook-addon-next/pull/5

Forgive its size (it includes the example repos that go along with it).

Among other refactors, this details the webpack magic approach I mentioned to get it working so that the developer doesn't have to reach into nextjs internals themselves to provide the addon the `RouterContext`

**Comment by lifeiscontent** [comment]
@RyanClementsHax I found it because I follow Micheal Shilman on Twitter. He was the one who convinced me to build the next router addon in the first place.

Im currently in the middle of launching my UI kit for a customer. https://lifetimeui.com but im happy to take a thorough look at the code once I wrap that up.

As for the PR I'm happy to accept upstream changes. I'm wondering how you solved the dependency problem with Contexts, from the 2 addons I maintain I've seen storybook clobber dependencies so when you try to bundle them into the addon they don't work which was why I went with the provider injection pattern. My original plan was to do what you're saying and I've since learned about the webpack magic you've mentioned and have planned to add that to my package, just haven't had free time recently.

In short, yes I'd be happy to accept a PR and will review the one you sent shortly.

Thanks for the interest in collaboration.

**Comment by RyanClementsHax** [comment]
@lifeiscontent I see you left comments on the pull request. I replied to them and am looking forward to your feedback. Is the general approach I took something you are comfortable including in your addon? I definitely agree it could use some massaging.

Also, I was thinking that the way forward could be the following:
1. We massage the implementation provided by this addon in this repo
2. I deploy those changes in this addon
3. If nothing breaks, we make a pull request pulling in those changes to your addon
4. You deploy those changes
5. If nothing breaks, we update this addon to package your addon
6. I deploy that change

What do you think?

**Comment by lifeiscontent** [comment]
@RyanClementsHax Yeah, I think that approach could work. Sorry I haven't followed up on comments feel free to move forward, as we get closer to opening a PR on my addon I'll do a deeper audit then. Unfortunately I don't have a ton of time on my hands right now, do you have discord or something you communicate on frequently? Happy to start chatting and figuring out how to collaborate on this.

**Comment by RyanClementsHax** [comment]
I emailed you my discord user
